Documentation/Evolution/RegexSyntax.md

```diff
@@ -0,0 +1,245 @@
+# Regular Expression Syntax
+
+- Authors: Hamish Knight, Michael Ilseman
+
+## Introduction
+
+We aim to parse a superset of the syntax accepted by a variety of popular regular expression engines.
+
+**TODO: Elaborate**
+
+## Engines supported
+
+We aim to implement a syntactic superset of:
+
+- [PCRE 2][pcre2-syntax], an "industry standard" of sorts, and a rough superset of Perl, Python, etc.
+- [Oniguruma][oniguruma-syntax], an internationalization-oriented engine with some modern features
+- [ICU][icu-syntax], used by NSRegularExpression, a Unicode-focused engine.
+- [.NET][.net-syntax]'s regular expressions, which support delimiter-balancing and some interesting minor details on conditional patterns.
+- **TODO: List Java here? It doesn't really add any more syntax than the above other than `\p{javaLowerCase}`**
+
+We also intend to achieve at least Level 1 (**TODO: do we want to promise Level 2?**) [UTS#18][uts18] conformance, which specifies regular expression matching semantics without mandating any particular syntax. However we can infer syntactic feature sets from its guidance.
+
+## Regex syntax supported
+
+### General syntax
+
+The following syntax are supported by all the above engines.
```

Comment:
I think this note is fine, though we may want to start shifting how we frame or talk about this into what Swift will support. It's super helpful to note and show what other engines support/do, especially since the main value add of the pitched syntax is as a superset.

Something more like we're formally proposing to add the following to Swift, with other engine notes along the way.